Fix TimelineSegmentGetter

AxelDelmas opened the pull request:
- simplify TimelineSegmentGetter logic
- Configurable maxSegmentAhead. ALways return full list of segment when index === -1 and requestedTime isn't specified
- Use time based segment search when available, and fall back on index based when it's not

Commit:
clean up method decideSegmentListRangeForTimeline

Commit:
Simplify SegmentTimeline logic

Commit:
RequestedMediaTime and requestedTime either defined or null

Commit:
Use requiredMediaTime if available, index otherwise

Commit:
cleaning up

ruslandinov commented on src/dash/utils/TimelineSegmentsGetter.js:
```diff
@@ -48,7 +48,13 @@ function TimelineSegmentsGetter(config, isDynamic) {
-        var maxSegmentsAhead = 10;
+        let maxSegmentsAhead;
+        if (availabilityUpperLimit) {
+            maxSegmentsAhead = availabilityUpperLimit
```
missing semicolon here

ruslandinov commented on src/dash/utils/TimelineSegmentsGetter.js:
```diff
@@ -48,7 +48,13 @@ function TimelineSegmentsGetter(config, isDynamic) {
+        if (availabilityUpperLimit) {
+            maxSegmentsAhead = availabilityUpperLimit
+        } else {
+            maxSegmentsAhead = (index > -1 || requestedTime !== null) ? 10 : Infinity;
```
dash.js is full of magic numbers, probably we should start fixing this by declaring constants, ie extracting `10` to a constant in this case.

ruslandinov reviewed (approved):
Seems good!

Commit:
Make sure requestedTime is always null when not specified.
Must use arguments with default at the end, but I'd rather not change this method signature.

Commit:
Fix lint

Commit:
fix jscs

ruslandinov commented on src/dash/utils/TimelineSegmentsGetter.js:
```diff
@@ -39,7 +39,12 @@ function TimelineSegmentsGetter(config, isDynamic) {
+
+        if (requestedTime === undefined) {
+            requestedTime = null;
+        }
```
👍 nice catch
